packages/core/src/browser/navigatable.ts

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (C) 2018 Red Hat, Inc. and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import URI from '../common/uri';
+
+/**
+ * Each widget which holds an uri to a workspace file and wants to be able to reveal that file in navigator,
+ * (e.g. editor, image viewer, diff editor, etc.) has to implement this interface and provide the file uri on demand.
+ * No additional registration is needed.
+ */
+export interface Navigatable {
+    readonly targetUri: URI | undefined;
```

Comment:
If we go with the nominal typing, can we have a method instead? I am not sure whether property getters can be overridden by clients. What's your take on that, @akosyakov?

Reply:
Yes, it would be fine with me to use `getTargetUri(): URI | undefined` as well.

Properties cannot be overriden, only implemented. One can always call a method from the getter in the implementation to overcome it.